lib/utils/logger.ts

```diff
@@ -64,6 +64,15 @@ class Logger {
 		this.formatMessage = logger.formatWithPrefix.bind(logger);
 	}
 
+	private static logger: Logger;
+
+	public static getLogger() {
+		if (!this.logger) {
+			this.logger = new Logger();
+		}
+		return this.logger;
+	}
```

Comment:
This new `getLogger()` method and its usage throughout CLI are in their own commit. These changes are an answer to the following warnings:
```
$ balena tunnel test-rpi -p 22222:43213
(node:43572) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 drain listeners added. Use emitter.setMaxListeners() to increase limit
(node:43572) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 error listeners added. Use emitter.setMaxListeners() to increase limit
(node:43572) MaxListenersExceededWarning: Possible EventEmitter memory leak detected. 11 close listeners added. Use emitter.setMaxListeners() to increase limit
```
I ran the CLI with warning stack traces enabled and the traces pointed to instances of Logger. It would have been the combination of Logger instances with other event listener usage elsewhere that would have combined to produce the warnings, so it's still that possible that a memory leak (dangling listeners) exist somewhere. But the static Logger change was sufficient to prevent the warning, hence at least mitigating the issue for now.